### DISCUSSION TITLE
node ./build/server/index.js error on Docker

### DISCUSSION THREAD

**Discussion by addvanced** [outline/outline#5466] [Self Hosting] [answered]
I have been running Outline Wiki in a Docker environment for quite some time. But after upgrading to v0.69.x, I have been running into an issue, where I am not able to start the container.

It keeps restarting, with the following error:
`yarn run v1.22.19
$ node ./build/server/index.js
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.`

I don't get much information from this error - and I am not a node/npm/yarn expert, so I am not quite sure how to debug this issue.

Anyone else had this problem?

I tried going back to the 0.68.1 image, and that seems to work fine, but of course now the database has been migrated to 0.69.2, and I would prefer making this work, than having to roll back the database and not upgrading to 0.69.x.

Any advice on how to debug the error that yarn / node is throwing, or what could be causing this issue would be appreciated.

**Comment by tommoor** [comment]
Did you want to post the server logs?

**Reply by addvanced** [reply]
What server logs exactly?
It's a docker container, using the official docker-compose example.

If you're talking about some server logs inside the docker container, then I would appreciate a hint where to find it. As mentioned, I am not a node/npm/yarn expert, and debugging these applications is a bit outside of my expertise.

The error posted in my question is everything that I see when I look at the docker container logs. It keeps repeating that error.
![outline_error](https://github.com/outline/outline/assets/37810041/97bcfee6-3c71-44c3-b840-16d49b03ad35)

**Reply by tommoor** [reply]
Unfortunately I'm not sure of any scenario where it would exit with an error code but not output any errors at all. You can try setting `DEBUG=*` environment variable and see if that opens anything more useful

**Reply by addvanced** [reply]
Sure, I will try that and get back to you!

**Reply by addvanced** [reply]
Here's a log with debugging enabled for everything. Hopefully this makes more sense to you than me. :)
[outline_logs.txt](https://github.com/outline/outline/files/11832965/outline_logs.txt)

**Reply by tommoor** [reply]
I'm at a loss, obviously we can see it's getting part way through the startup process but not very far at all and then exiting with an error code.

In the source there are 6 places where we exit with code "1" (`process.exit(1)`) and all of them have a log before explaining the error which would appear alongside here.

**Reply by addvanced** [reply]
Yup. I have been looking through the source as well, and it doesn't make sense to me either.

I have tried to run this docker container on 2 different machines (Ubuntu Server 22.04) and the official installation of Docker Engine as described on their website.

And as I mentioned, it worked fine before I upgraded to 0.69.x. So it shouldn't have anything to do with my server environment, docker-compose file or docker.env file.

I will try and look into it some more. Maybe start up a completely "clean" install of 0.69.2, with a new database etc.

**Reply by addvanced** [reply]
Okay. I might have found the error.
Maybe you can confirm if any changes has been made to this since 0.68.1.

In my docker.env i previously had LOG_LEVEL=DEBUG.
This worked in 0.68.1, but it causes the error above in 0.69.x.

Why? Because DEBUG is uppercase. Seems like 0.68.x is case-insensitive, where as 0.69.x is NOT.
Changing this to LOG_LEVEL=debug solved the problem for me.

**Reply by tommoor** [reply, marked as answer]
It was changed here: https://github.com/outline/outline/pull/4937

There is validation – but of course if the logging itself is broken then the validation errors won't appear 🤦

**Reply by addvanced** [reply]
Awesome! 
Then problem solved - and everything is working! yiha! Back to writing notes!

Thanks a lot for your time! :)

**Reply by tommoor** [reply]
Fixed for the future:
https://github.com/outline/outline/commit/831318d94137331b382d3dd9710dee62c9ae67e3

**Reply by addvanced** [reply]
Hurrayy! Awesome!
